x-pack/plugins/actions/server/plugin.ts

```diff
@@ -343,6 +356,14 @@ export class ActionsPlugin implements Plugin<PluginSetupContract, PluginStartCon
         ensureSufficientLicense(actionType);
         actionTypeRegistry.register(actionType);
       },
+      registerSubActionConnectorType: <
+        Config extends ActionTypeConfig = ActionTypeConfig,
+        Secrets extends ActionTypeSecrets = ActionTypeSecrets
+      >(
+        connector: SubActionConnectorType<Config, Secrets>
+      ) => {
+        subActionFramework.registerConnector(connector);
```

Comment:
> do we neeed ensureSufficientLicense(actionType) here, or similar? Looks like we don't, because eventually registerType() will get called for this connector type, with the constructed args.

That is correct!

> I found myself wondering this kind of thing looking through the other bits here. Does the executor do all the things the framework's executor does, like log events? Yes, the framework connector does all the usual stuff, and calls the subaction connector executor in the usual way, but that subaction connector makes a call based on the subAction. Perhaps we should put something in the README showing the flow through the executor.

Yes, this framework constructs a common executor for all connectors which is passed as the actual executor to the actions' framework. Do you want me to add a diagram about the flow of the common executor (as I have in the description of the PR) or another diagram that shows that the constructed common executor is given to the actions' framework as the executor function?

Reply:
I think what I was missing was how the connector itself would be registered, and then presumably also the subcommands are registered at the same time.  So, code would have helped :-).  Seeing that would have firmed this stuff up for me.  And per my note re: function tests, I guess it would be nice to have one, which was basically a demo of a new test connector with subcommands registered for it.  Then you can just a link to the top comment for that bit of code :-)

Reply:
🙂  I updated the docs accordingly in this commit [`7255e8a` (#129307)](https://github.com/elastic/kibana/pull/129307/commits/7255e8aa7795206655efcd2cfc1d9460bc2f4d2c). Let me know if it is not sufficient.